test/e2e/deployment/deployment.go

```diff
@@ -0,0 +1,41 @@
+package deployment
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	deputils "k8s.io/kubectl/pkg/util/deployment"
+)
+
+type DeploymentCommands struct{ deployment appsv1.Deployment }
```

Comment:
Do same as DeamonSet here creating a custom matcher and storing it at `matcher.go`

Users should do:

```golang
deployment.Eventually(DefaultNMStateWebhook).Should(deployment.BeReady())
deployment.Eventually(DefaultNMStateWebhook).ShouldNot(deployment.BeReady())
```

Reply:
@bcrochet are you ok doing the Deployment matcher ?

Reply:
Yes. Should be done.